.github/workflows/backport.yml

```diff
@@ -0,0 +1,28 @@
+name: Backport Merged Pull Request
+
+on:
+  pull_request:
+    types:
+      - closed
```

Comment:
I have updated the backport.yml to trigger the backport script when a pull request is closed or labeled and pull request should fulfill following condition:
1. Pull request should be closed
2. Pull request should be merged
3. Pill request should have a label containing `backport` string.

Reply:
Below are the improvements that can be made to ensure the script's idempotence:

1. Validate whether the branch exists before creating it.
2. Check if the pull request already exists before attempting to create it.
3. Verify if a comment has been added to the pull request before attempting to add another.
4. Validate whether the branch already contains cherry-picked commits.

All validations are straightforward except for validating cherry-picked commits. If you have any suggestions on how to perform this validation, please let me know.

Reply:
I have implemented the suggested checks. Specifically, I added a check for branch existence; if the branch exists, it deletes the branch with all temporary changes, recreates it, and restarts the backport process.

I chose not to use `git rebase` because it is more error-prone and can potentially cause conflicts or unintended changes. Instead, I opted for branch deletion and recreation to ensure a clean and stable backport process.

Reply:
Additionally, I included a check to verify if a pull request exists between the target branch and the new backport branch. If a pull request exists, the backport process is skipped; otherwise, it continues. 
I also added a manual test case as suggested.
PTAL again.